src/main/java/com/luma/drivers/WebDriverFactory.java

```diff
@@ -0,0 +1,100 @@
+package com.luma.drivers;
+
+import org.apache.log4j.Logger;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.remote.AbstractDriverOptions;
+import org.openqa.selenium.safari.SafariDriver;
+import org.openqa.selenium.safari.SafariOptions;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.Optional;
+import java.util.Properties;
+
+public class WebDriverFactory {
```

Comment:
Leave findDriver, getDriver, getDriver in WebDriverFactory, another logic move in a separate class, read about single responsible principle